File: python/exercice1-1-3/palyndrome.js
function estPalindrome(str) {
    var strTmp = str.trim();
    var spaceLeft= 0, spaceRight = 0;
    for (var i = 0; i < strTmp.length / 2; i++) {
        while (strTmp[i + spaceLeft] === ' ') {
            spaceLeft++;
        }
        while (strTmp[strTmp.length - i - spaceRight - 1] === ' ') {
            spaceRight++;
        }
        if(strTmp[i + spaceLeft] !== strTmp[strTmp.length - i - spaceRight - 1]) {
            return false;
        }
      }

    return true;
}

var str = "       Hello World!        ";
console.log(str);
var trimmedStr = str.trim();
console.log(trimmedStr);

console.log(estPalindrome("ESOPE RESTE ICI ET SE   REPOE"));

File: python/exercice1-1-3/palyndrome.py
def estPalindrome(string):
    strTmp = string.replace(" ", "");
    for index in range(int(len(strTmp) / 2)):
        if strTmp[index] != strTmp[len(strTmp) - index - 1]:
            return False;
    return True;

str = "ABCd"
print(str[::-1])

print(len("ESOPE RESTE ICI ET SE REPOSE") / 2)
print(estPalindrome("ESOPE RESTE ICI ET SE REPOSE"))
print(estPalindrome("YOOYP"))